src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -661,21 +681,24 @@ class WindowsPtyHeuristics extends Disposable {
 				if (prompt) {
 					const adjustedPrompt = typeof prompt === 'string' ? prompt : prompt.prompt;
 					this._capability.currentCommand.commandStartMarker = this._terminal.registerMarker(0)!;
-					if (typeof prompt === 'object' && prompt.likelySingleLine) {
-						this._logService.debug('CommandDetectionCapability#_tryAdjustCommandStartMarker adjusted promptStart', `${this._capability.currentCommand.promptStartMarker?.line} -> ${this._capability.currentCommand.commandStartMarker.line}`);
-						this._capability.currentCommand.promptStartMarker?.dispose();
-						this._capability.currentCommand.promptStartMarker = cloneMarker(this._terminal, this._capability.currentCommand.commandStartMarker);
-						// Adjust the last command if it's not in the same position as the following
-						// prompt start marker
-						const lastCommand = this._capability.commands.at(-1);
-						if (lastCommand && this._capability.currentCommand.commandStartMarker.line !== lastCommand.endMarker?.line) {
-							lastCommand.endMarker?.dispose();
-							lastCommand.endMarker = cloneMarker(this._terminal, this._capability.currentCommand.commandStartMarker);
-						}
+
```

Comment:
This is a no go imo, to give you an idea of what I see here:

Before:

![Recording 2024-04-01 at 04 26 04](https://github.com/microsoft/vscode/assets/2193314/76cdc4e0-8a7b-4b72-9cc1-3d52ba606d2d)

After:

![Recording 2024-04-01 at 04 26 44](https://github.com/microsoft/vscode/assets/2193314/02a3f0cd-c737-40a0-96ab-0d3011a72aea)

It's a little hard to see in the gif but it also drives other smaller things like the grey bar on hover:

![image](https://github.com/microsoft/vscode/assets/2193314/1a0fb047-3216-4471-be89-dc4fee36905c)

The blue bar on command nav (ctrl+up):

![image](https://github.com/microsoft/vscode/assets/2193314/e23f43a3-969f-49ac-9f37-d2620c324f61)

(very recent change) Also in run recent command (ctrl+alt+r):

![image](https://github.com/microsoft/vscode/assets/2193314/3c192fd4-5ce5-4a34-a900-23aaaadfe767)

---

Is there any way we can get the best of both worlds here? For example we could customize the `_tryAdjustCommandStartMarker` behavior when it comes via `handleInput`?

Reply:
@Tyriar The before and after gifs look very similar to me at this speed, what is the major difference between the two? Happy to adjust the `_tryAdjustCommandStartMarker` for each case

Reply:
The major difference is that the powerline part is included in the sticky scroll too, but importantly not the initial empty line of the prompt.

Reply:
I adjusted the logic to actually pull the prompt height during the prompt, similar to the `cwd` to allow for proper prompt adjustment and it seems to be working much better for the multiline prompts. Do you think this is an acceptable approach? I'm occasionally getting left over markers with single line powershell that I can work on if you like it

## Working Examples
https://github.com/microsoft/vscode/assets/35637443/2190db89-5010-42ea-af22-eed77729c570


https://github.com/microsoft/vscode/assets/35637443/5d85ab16-fcf9-4ec9-baa2-6b6586a69d0b